thebearingedge/node-tinyhttp-realworld-example-app
# Created articles & comments routes

### Some recap of what I did:

Routes I created:
* `GET /api/articles`
* `DELETE /api/articles/:slug`
* `GET /api/articles/:slug/comments`
* `POST /api/articles/:slug/comments`
* `DELETE /api/articles/:slug/comments/:id`

Tests I created:
* delete article
* add comments
* get comments (incomplete)
* delete comments (incomplete)

Documentations:
* README.md (we should edit this one along the way, as the repo progresses)
* LICENSE
* CONTRIBUTING.md (please check this one if I got everything right)

Others:
* `.nvmrc` file to integrate with [nvm](https://github.com/nvm-sh/nvm) or [fnm](https://github.com/Schniz/fnm)
* `.github/workflows/test.yml` for CI testing. after the merge, you should uncomment the codecov part. But don't forget to make this repo available on codecov first.

### Question

This is my first time trying Prisma, so I kind of confused about getting the relation between user and following/followers. So on a few files, I commented out the `following` part. Like this: 

```js
/api/article/article-routes.js
const found = await prisma.article.findMany({
      ...
      select: {
        ...
        // @TODO: If a users is logged in, check whether or not they favorited this article.
        // favorited: boolean
        author: {
          select: {
            username: true,
            bio: true,
            image: true
            // @TODO: If a users is logged in, check whether or not they follow this author.
            // following: boolean
          }
        }
      }
    })
```

How would you approach that part? I thought earlier that a new middleware (ex. `optionalAuth`) needs to be created to supply some data, so we can put it on the query.
